### PR TITLE
Adding office-runtime type

### PR DESCRIPTION
**Pull request by mscharlock**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/office-runtime/tslint.json [hunk 1]
```diff
@@ -0,0 +1,79 @@
+{
```
Comment (RyanCavanaugh) [inline types/office-runtime/tslint.json]: Please only disable the lint rules that you have to. It looks like you already have per-line lint rule excludes in the definition file -- prefer those over bulk disables whenever practical. Thanks!

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]

File: types/office-runtime/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+       * Callback that is run when the dialog box is closed.
+       *
+       * 
+       */
+      onClose?: () => void;
+  
+      /**
+       * @beta
+       * Callback that is run when the dialog box sends an error.
+       *
```
Comment (SimonSchick) [inline types/office-runtime/index.d.ts]: You can ditch these empty lines everywhere.

File: types/office-runtime/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+       */
+      hideTitle?: boolean;
+  
+      /**
+       * @beta
+       * Callback that is run when the dialog box sends a message to its parent.
+       *
+       *
+       */
+      onMessage?: (message: string, dialog?: Dialog) => void;
```
Comment (SimonSchick) [inline types/office-runtime/index.d.ts]: the short form is preferred iirc eg. `onMessage?(message: string, dialog?: Dialog): void`

File: types/office-runtime/office-runtime-tests.ts [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    }
+)
+
+
+OfficeRuntime.displayWebDialog("https://localhost:3000", { 
+    height: '50', 
+    width: '50%',
+    displayInIFrame: false,
+    hideTitle: true, 
+        onMessage: (message, dialog) => {
```
Comment (SimonSchick) [inline types/office-runtime/office-runtime-tests.ts]: Nit: Formatting.

File: types/office-runtime/tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,23 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6", 
+            "dom"
```
Comment (SimonSchick) [inline types/office-runtime/tsconfig.json]: I skipped over this code, do we need `dom` here?

File: types/office-runtime/tslint.json (hunk 1 above)
Comment (SimonSchick) [reply types/office-runtime/tslint.json]: Yea, please don't copy/paste from other packages, see the `readme.md` for more info.

**Review by SimonSchick — CHANGES_REQUESTED** [review]
Please correct the lint config and run `npm run lint office-runtime`, I see quiet a few violations.

File: types/office-runtime/index.d.ts [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+  
+      /*
+       * @beta
+       * Callback that is run when the dialog box is closed.
+       *
```
Comment (SimonSchick) [inline types/office-runtime/index.d.ts]: Also here, please try to purge all of those wasted lines, nicer to read for the consumer :)

File: types/office-js/index.d.ts [hunk 7]
```diff
@@ -17176,9 +17176,12 @@ declare namespace OfficeExtension {
 ////////////////////////////////////////////////////////////////
 
 
+
```
[the remark this answers is not in the document]
Comment (SimonSchick) [reply types/office-js/index.d.ts]: It's generally a bad idea for this repo to touch any typings that are out of scope of the PR as this pings extra contributors and makes the build process take longer.

**Review by SimonSchick — APPROVED** [review]

**Pull request merged**